### ISSUE TITLE
[5.x] AppImage fails to start on debian trixie

### ISSUE THREAD

**Issue by t-cc** [Ultimaker/Cura#19837]
### Cura Version

5.8.1

### Operating System

Linux debian trixie, kde desktop

### Printer

all printers

### Reproduction steps

Try to start cura from the App Image file on debian trixie

### Actual results

```
qt.qpa.gl: Xcb EGL gl-integration initialize failed
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
Could not initialize GLX
Abortado (`core' generado)
```

### Expected results

Cura starts

### Add your .zip and screenshots here ⬇️

[stderr.log](https://github.com/user-attachments/files/17576995/stderr.log)

**Comment by ohagandavid** [comment]
I have this issue on a trixie install too. 5.6.0 and 5.8.1 appImages affected. Apt logs have a bunch of updates to qt6 on 10/26. libqt6opengl6?

**Comment by witenite** [comment]
I am having the same issue on Trixie, with 5.8.1 and 5.9.0-Beta1

As a normal user and under Sudo....

**Comment by susno762** [comment]
Same issue here with Debian Trixie and 5.8.1. Output of --debug attached.

[debug.log](https://github.com/user-attachments/files/17610932/debug.log)

**Comment by TheRealOne78** [comment]
This seems to happen only on AMD GPUs, not necessarily just on trixie.

This happens even on the older `modern` builds

**Comment by JeanLucCoulon** [comment]
I've an Intel CPU (core I7 6700K) with an integrated GPU (HD Graphics 530),
and I've the problem with debian sid.
It used to work (Ultimaker Cura 5.8.0) on the same hardware
Regards
Jean-Luc

Le mar. 5 nov. 2024 à 21:26, TheRealOne78 ***@***.***> a
écrit :

**Comment by madebyTimo** [comment]
I also have the problem on Intel GPU (8. gen CPU with Intel UHD Graphics 620).
On Nvidia GPU (RTX 3070) with proprietary drivers it works fine.

Both systems have latest Debian Trixie. The problem startet with an recent apt upgrade where some libqt6 packages were updated.

**Comment by SpyroTEQ** [comment]
I just made an `apt dist-upgrade` on my Kali machine (so Debian based too), and got a Python crash on starting cura 5.0.0-3 (the software stats, but crashes right away):
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/UM/Qt/ListModel.py", line 54, in data
    return self._items[index.row()][self._role_names[role].decode("utf-8")]
           ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'material_name'
```

So I turned to latest AppImage, and got the same issue as described here (Could not initialize GLX) on a core2duo with a HD6770:
```
┌──(xenos㉿homefixe)-[~/Téléchargements]
└─$ md5sum ./UltiMaker-Cura-5.8.1-linux-X64.AppImage
16bd0c4a7afcb60f08b0da882095809d  ./UltiMaker-Cura-5.8.1-linux-X64.AppImage
                                                                                                                                                                                                                                              
┌──(xenos㉿homefixe)-[~/Téléchargements]
└─$ ./UltiMaker-Cura-5.8.1-linux-X64.AppImage       
Gtk-Message: 21:46:24.039: Failed to load module "canberra-gtk-module"
Gtk-Message: 21:46:24.044: Failed to load module "canberra-gtk-module"
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
Could not initialize GLX
zsh: IOT instruction  ./UltiMaker-Cura-5.8.1-linux-X64.AppImage
                                                                                                                                                                                                                                              
┌──(xenos㉿homefixe)-[~/Téléchargements]
└─$ cat /proc/cpuinfo 
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 15
model name	: Intel(R) Core(TM)2 CPU          6600  @ 2.40GHz
stepping	: 6
microcode	: 0xd0
cpu MHz		: 2404.075
cache size	: 4096 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 2
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 10
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl cpuid aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm lahf_lm pti tpr_shadow dtherm
vmx flags	: tsc_offset vtpr
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit mmio_unknown
bogomips	: 4808.15
clflush size	: 64
cache_alignment	: 64
address sizes	: 36 bits physical, 48 bits virtual
power management:

processor	: 1
vendor_id	: GenuineIntel
cpu family	: 6
model		: 15
model name	: Intel(R) Core(TM)2 CPU          6600  @ 2.40GHz
stepping	: 6
microcode	: 0xd0
cpu MHz		: 2404.071
cache size	: 4096 KB
physical id	: 0
siblings	: 2
core id		: 1
cpu cores	: 2
apicid		: 1
initial apicid	: 1
fpu		: yes
fpu_exception	: yes
cpuid level	: 10
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl cpuid aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm lahf_lm pti tpr_shadow dtherm
vmx flags	: tsc_offset vtpr
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit mmio_unknown
bogomips	: 4808.15
clflush size	: 64
cache_alignment	: 64
address sizes	: 36 bits physical, 48 bits virtual
power management:

                                                                                                                                                                                                                                              
┌──(xenos㉿homefixe)-[~/Téléchargements]
└─$ sudo lshw -C display
  *-display                 
       description: VGA compatible controller
       produit: Juniper XT [Radeon HD 6770]
       fabriquant: Advanced Micro Devices, Inc. [AMD/ATI]
       identifiant matériel: 0
       information bus: pci@0000:08:00.0
       nom logique: /dev/fb0
       version: 00
       bits: 64 bits
       horloge: 33MHz
       fonctionnalités: pm pciexpress msi vga_controller bus_master cap_list rom fb
       configuration: depth=32 driver=radeon latency=0 resolution=1920,1080
       ressources: irq:28 mémoire:d0000000-dfffffff mémoire:ff9e0000-ff9fffff portE/S:c800(taille=256) mémoire:c0000-dffff
                                                                                                                                                                                                                                              
┌──(xenos㉿homefixe)-[~/Téléchargements]
└─$ lsb_release -a                     
No LSB modules are available.
Distributor ID:	Kali
Description:	Kali GNU/Linux Rolling
Release:	2024.3
Codename:	kali-rolling
```

--debug shows:
```
┌──(xenos㉿homefixe)-[~/Téléchargements]
└─$ ./UltiMaker-Cura-5.8.1-linux-X64.AppImage --debug
trimesh/curvature.py:12: DeprecationWarning: Please use `coo_matrix` from the `scipy.sparse` namespace, the `scipy.sparse.coo` namespace is deprecated.
  from scipy.sparse.coo import coo_matrix
Gtk-Message: 21:52:15.665: Failed to load module "canberra-gtk-module"
Gtk-Message: 21:52:15.667: Failed to load module "canberra-gtk-module"
[MainThread] UM.Application.initialize [163]: Initializing UltiMaker Cura
[MainThread] UM.Application.initialize [164]: App Version 5.8.1
[MainThread] UM.Application.initialize [165]: Api Version 8.8.0
[MainThread] UM.Application.initialize [166]: Build type None
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaVHV0x7/../Resources/share/uranium/plugins must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaVHV0x7/../Resources/share/cura/plugins must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib/uranium must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib64/uranium must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib32/uranium must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaVHV0x7/plugins must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/Resources/uranium/plugins must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/Resources/cura/plugins must be a folder.
[MainThread] UM.Resources.__initializeStoragePaths [481]: Initializing storage paths
[MainThread] UM.Resources.__initializeStoragePaths [491]: Config storage path is /home/xenos/.config/cura/5.8
[MainThread] UM.Resources.__initializeStoragePaths [499]: Data storage path is /home/xenos/.local/share/cura/5.8
[MainThread] UM.Resources.__initializeStoragePaths [511]: Cache storage path is /home/xenos/.cache/cura/5.8
UM/Qt/QtApplication.py:701: DeprecationWarning: QEvent constructor is deprecated
  super().__init__(self.QtFunctionEvent)
[MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaVHV0x7/share/uranium/resources/bundled_packages/uranium.json
[MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaVHV0x7/share/cura/resources/bundled_packages/bundled_CuraEngineGradualFlow.json
[MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaVHV0x7/share/cura/resources/bundled_packages/cura.json
[MainThread] UM.View.GL.OpenGLContext.detectBestOpenGLVersion [124]: Trying OpenGL context 4.1...
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
Could not initialize GLX
Fatal Python error: Aborted

Current thread 0x00007efe75526740 (most recent call first):
  File "UM/View/GL/OpenGLContext.py", line 36 in setContext
  File "UM/View/GL/OpenGLContext.py", line 125 in detectBestOpenGLVersion
  File "UM/Qt/QtApplication.py", line 178 in initialize
  File "cura/CuraApplication.py", line 348 in initialize
  File "UM/Application.py", line 288 in run
  File "UM/Qt/QtApplication.py", line 385 in run
  File "cura/CuraApplication.py", line 880 in run
  File "cura_app.py", line 243 in <module>

Extension modules: PyQt6.QtCore, PyQt6.QtNetwork, PyQt6.QtGui, PyQt6.QtWidgets, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg.lapack_lite, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, scipy._lib._ccallback_c, scipy.sparse._sparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._isolve._iterative, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg._cythonized_array_utils, scipy.linalg._flinalg, scipy.linalg._solve_toeplitz, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg.cython_lapack, scipy.linalg.cython_blas, scipy.linalg._matfuncs_expm, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.special._ellip_harm_2, scipy.spatial.transform._rotation, _cffi_backend, PyQt6.QtOpenGL, PyQt6.QtQml, PyQt6.QtQuick, pyArcus, PyQt6.QtSvg, pyclipper._pyclipper, pynest2d, scipy.optimize._minpack2, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize.__nnls, scipy.optimize._highs.cython.src._highs_wrapper, scipy.optimize._highs._highs_wrapper, scipy.optimize._highs.cython.src._highs_constants, scipy.optimize._highs._highs_constants, scipy.linalg._interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, shapely.lib, shapely._geos, shapely._geometry_helpers, scipy.ndimage._nd_image, _ni_label, scipy.ndimage._ni_label (total: 87)
zsh: IOT instruction  ./UltiMaker-Cura-5.8.1-linux-X64.AppImage --debug
```
Same with `./UltiMaker-Cura-5.9.0-beta.2-linux-X64.AppImage` and weirdly, with `Ultimaker-Cura-5.0.0-linux.AppImage` too (to 5.0.0-3 from apt boots then crashes, but the AppImage in the releases crashes before any screen shows up).

I kept trying older versions, and the `Ultimaker_Cura-4.13.0.AppImage` is working.

I hope that helps

**Comment by witenite** [comment]
> I just made an `apt dist-upgrade` on my Kali machine (so Debian based too), and got a Python crash on starting cura 5.0.0-3 (the software stats, but crashes right away):
> 
> ```
> Traceback (most recent call last):
>   File "/usr/lib/python3/dist-packages/UM/Qt/ListModel.py", line 54, in data
>     return self._items[index.row()][self._role_names[role].decode("utf-8")]
>            ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> KeyError: 'material_name'
> ```
> 
> So I turned to latest AppImage, and got the same issue as described here (Could not initialize GLX) on a core2duo with a HD6770:
> 
> ```
> ┌──(xenos㉿homefixe)-[~/Téléchargements]
> └─$ md5sum ./UltiMaker-Cura-5.8.1-linux-X64.AppImage
> 16bd0c4a7afcb60f08b0da882095809d  ./UltiMaker-Cura-5.8.1-linux-X64.AppImage
>                                                                                                                                                                                                                                               
> ┌──(xenos㉿homefixe)-[~/Téléchargements]
> └─$ ./UltiMaker-Cura-5.8.1-linux-X64.AppImage       
> Gtk-Message: 21:46:24.039: Failed to load module "canberra-gtk-module"
> Gtk-Message: 21:46:24.044: Failed to load module "canberra-gtk-module"
> qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
> qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
> Could not initialize GLX
> zsh: IOT instruction  ./UltiMaker-Cura-5.8.1-linux-X64.AppImage
>                                                                                                                                                                                                                                               
> ┌──(xenos㉿homefixe)-[~/Téléchargements]
> └─$ cat /proc/cpuinfo 
> processor	: 0
> vendor_id	: GenuineIntel
> cpu family	: 6
> model		: 15
> model name	: Intel(R) Core(TM)2 CPU          6600  @ 2.40GHz
> stepping	: 6
> microcode	: 0xd0
> cpu MHz		: 2404.075
> cache size	: 4096 KB
> physical id	: 0
> siblings	: 2
> core id		: 0
> cpu cores	: 2
> apicid		: 0
> initial apicid	: 0
> fpu		: yes
> fpu_exception	: yes
> cpuid level	: 10
> wp		: yes
> flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl cpuid aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm lahf_lm pti tpr_shadow dtherm
> vmx flags	: tsc_offset vtpr
> bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit mmio_unknown
> bogomips	: 4808.15
> clflush size	: 64
> cache_alignment	: 64
> address sizes	: 36 bits physical, 48 bits virtual
> power management:
> 
> processor	: 1
> vendor_id	: GenuineIntel
> cpu family	: 6
> model		: 15
> model name	: Intel(R) Core(TM)2 CPU          6600  @ 2.40GHz
> stepping	: 6
> microcode	: 0xd0
> cpu MHz		: 2404.071
> cache size	: 4096 KB
> physical id	: 0
> siblings	: 2
> core id		: 1
> cpu cores	: 2
> apicid		: 1
> initial apicid	: 1
> fpu		: yes
> fpu_exception	: yes
> cpuid level	: 10
> wp		: yes
> flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl cpuid aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm lahf_lm pti tpr_shadow dtherm
> vmx flags	: tsc_offset vtpr
> bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit mmio_unknown
> bogomips	: 4808.15
> clflush size	: 64
> cache_alignment	: 64
> address sizes	: 36 bits physical, 48 bits virtual
> power management:
> 
>                                                                                                                                                                                                                                               
> ┌──(xenos㉿homefixe)-[~/Téléchargements]
> └─$ sudo lshw -C display
>   *-display                 
>        description: VGA compatible controller
>        produit: Juniper XT [Radeon HD 6770]
>        fabriquant: Advanced Micro Devices, Inc. [AMD/ATI]
>        identifiant matériel: 0
>        information bus: pci@0000:08:00.0
>        nom logique: /dev/fb0
>        version: 00
>        bits: 64 bits
>        horloge: 33MHz
>        fonctionnalités: pm pciexpress msi vga_controller bus_master cap_list rom fb
>        configuration: depth=32 driver=radeon latency=0 resolution=1920,1080
>        ressources: irq:28 mémoire:d0000000-dfffffff mémoire:ff9e0000-ff9fffff portE/S:c800(taille=256) mémoire:c0000-dffff
>                                                                                                                                                                                                                                               
> ┌──(xenos㉿homefixe)-[~/Téléchargements]
> └─$ lsb_release -a                     
> No LSB modules are available.
> Distributor ID:	Kali
> Description:	Kali GNU/Linux Rolling
> Release:	2024.3
> Codename:	kali-rolling
> ```
> 
> --debug shows:
> 
> ```
> ┌──(xenos㉿homefixe)-[~/Téléchargements]
> └─$ ./UltiMaker-Cura-5.8.1-linux-X64.AppImage --debug
> trimesh/curvature.py:12: DeprecationWarning: Please use `coo_matrix` from the `scipy.sparse` namespace, the `scipy.sparse.coo` namespace is deprecated.
>   from scipy.sparse.coo import coo_matrix
> Gtk-Message: 21:52:15.665: Failed to load module "canberra-gtk-module"
> Gtk-Message: 21:52:15.667: Failed to load module "canberra-gtk-module"
> [MainThread] UM.Application.initialize [163]: Initializing UltiMaker Cura
> [MainThread] UM.Application.initialize [164]: App Version 5.8.1
> [MainThread] UM.Application.initialize [165]: Api Version 8.8.0
> [MainThread] UM.Application.initialize [166]: Build type None
> [MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaVHV0x7/../Resources/share/uranium/plugins must be a folder.
> [MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaVHV0x7/../Resources/share/cura/plugins must be a folder.
> [MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib/uranium must be a folder.
> [MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib64/uranium must be a folder.
> [MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib32/uranium must be a folder.
> [MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaVHV0x7/plugins must be a folder.
> [MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/Resources/uranium/plugins must be a folder.
> [MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/Resources/cura/plugins must be a folder.
> [MainThread] UM.Resources.__initializeStoragePaths [481]: Initializing storage paths
> [MainThread] UM.Resources.__initializeStoragePaths [491]: Config storage path is /home/xenos/.config/cura/5.8
> [MainThread] UM.Resources.__initializeStoragePaths [499]: Data storage path is /home/xenos/.local/share/cura/5.8
> [MainThread] UM.Resources.__initializeStoragePaths [511]: Cache storage path is /home/xenos/.cache/cura/5.8
> UM/Qt/QtApplication.py:701: DeprecationWarning: QEvent constructor is deprecated
>   super().__init__(self.QtFunctionEvent)
> [MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaVHV0x7/share/uranium/resources/bundled_packages/uranium.json
> [MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaVHV0x7/share/cura/resources/bundled_packages/bundled_CuraEngineGradualFlow.json
> [MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaVHV0x7/share/cura/resources/bundled_packages/cura.json
> [MainThread] UM.View.GL.OpenGLContext.detectBestOpenGLVersion [124]: Trying OpenGL context 4.1...
> qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
> qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
> Could not initialize GLX
> Fatal Python error: Aborted
> 
> Current thread 0x00007efe75526740 (most recent call first):
>   File "UM/View/GL/OpenGLContext.py", line 36 in setContext
>   File "UM/View/GL/OpenGLContext.py", line 125 in detectBestOpenGLVersion
>   File "UM/Qt/QtApplication.py", line 178 in initialize
>   File "cura/CuraApplication.py", line 348 in initialize
>   File "UM/Application.py", line 288 in run
>   File "UM/Qt/QtApplication.py", line 385 in run
>   File "cura/CuraApplication.py", line 880 in run
>   File "cura_app.py", line 243 in <module>
> 
> Extension modules: PyQt6.QtCore, PyQt6.QtNetwork, PyQt6.QtGui, PyQt6.QtWidgets, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg.lapack_lite, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, scipy._lib._ccallback_c, scipy.sparse._sparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._isolve._iterative, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg._cythonized_array_utils, scipy.linalg._flinalg, scipy.linalg._solve_toeplitz, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg.cython_lapack, scipy.linalg.cython_blas, scipy.linalg._matfuncs_expm, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.special._ellip_harm_2, scipy.spatial.transform._rotation, _cffi_backend, PyQt6.QtOpenGL, PyQt6.QtQml, PyQt6.QtQuick, pyArcus, PyQt6.QtSvg, pyclipper._pyclipper, pynest2d, scipy.optimize._minpack2, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize.__nnls, scipy.optimize._highs.cython.src._highs_wrapper, scipy.optimize._highs._highs_wrapper, scipy.optimize._highs.cython.src._highs_constants, scipy.optimize._highs._highs_constants, scipy.linalg._interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, shapely.lib, shapely._geos, shapely._geometry_helpers, scipy.ndimage._nd_image, _ni_label, scipy.ndimage._ni_label (total: 87)
> zsh: IOT instruction  ./UltiMaker-Cura-5.8.1-linux-X64.AppImage --debug
> ```
> 
> Same with `./UltiMaker-Cura-5.9.0-beta.2-linux-X64.AppImage` and weirdly, with `Ultimaker-Cura-5.0.0-linux.AppImage` too (to 5.0.0-3 from apt boots then crashes, but the AppImage in the releases crashes before any screen shows up).
> 
> I kept trying older versions, and the `Ultimaker_Cura-4.13.0.AppImage` is working.
> 
> I hope that helps

My debug results looks pretty much the same.

**Comment by boorcecom** [comment]
Same here on Debian trixie on a Lenovo Laptop.
Ryzen 7 5800H (amggpu) + Nvidia RTX4060, using wayland.
1st, need to run with QT_QPA_PLATFORM=xcb (to avoid an error with wayland...)

But it fails with the same error.
`QT_QPA_PLATFORM=xcb ./UltiMaker-Cura-5.8.1-linux-X64.AppImage --debug
trimesh/curvature.py:12: DeprecationWarning: Please use `coo_matrix` from the `scipy.sparse` namespace, the `scipy.sparse.coo` namespace is deprecated.
  from scipy.sparse.coo import coo_matrix
Gtk-Message: 11:34:35.056: Failed to load module "canberra-gtk-module"
Gtk-Message: 11:34:35.056: Failed to load module "canberra-gtk-module"
[MainThread] UM.Application.initialize [163]: Initializing UltiMaker Cura
[MainThread] UM.Application.initialize [164]: App Version 5.8.1
[MainThread] UM.Application.initialize [165]: Api Version 8.8.0
[MainThread] UM.Application.initialize [166]: Build type None
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaANLMru/../Resources/share/uranium/plugins must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaANLMru/../Resources/share/cura/plugins must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib/uranium must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib64/uranium must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/lib32/uranium must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/.mount_UltiMaANLMru/plugins must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/Resources/uranium/plugins must be a folder.
[MainThread] UM.PluginRegistry.addPluginLocation [205]: Plugin location /tmp/Resources/cura/plugins must be a folder.
[MainThread] UM.Resources.__initializeStoragePaths [481]: Initializing storage paths
[MainThread] UM.Resources.__initializeStoragePaths [491]: Config storage path is /home/ncazier@ad.boorce.com/.config/cura/5.8
[MainThread] UM.Resources.__initializeStoragePaths [499]: Data storage path is /home/ncazier@ad.boorce.com/.local/share/cura/5.8
[MainThread] UM.Resources.__initializeStoragePaths [511]: Cache storage path is /home/ncazier@ad.boorce.com/.cache/cura/5.8
[MainThread] UM.Qt.QtApplication.initialize [164]: Preferences file not found, ignore and use default language 'en_US'
[MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaANLMru/share/uranium/resources/bundled_packages/uranium.json
[MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaANLMru/share/cura/resources/bundled_packages/bundled_CuraEngineGradualFlow.json
[MainThread] UM.PackageManager.__init__ [56]: Found bundled packages JSON file: /tmp/.mount_UltiMaANLMru/share/cura/resources/bundled_packages/cura.json
[MainThread] UM.View.GL.OpenGLContext.detectBestOpenGLVersion [124]: Trying OpenGL context 4.1...
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
Could not initialize GLX
Fatal Python error: Aborted

Current thread 0x00007f74d82c2740 (most recent call first):
  File "UM/View/GL/OpenGLContext.py", line 36 in setContext
  File "UM/View/GL/OpenGLContext.py", line 125 in detectBestOpenGLVersion
  File "UM/Qt/QtApplication.py", line 178 in initialize
  File "cura/CuraApplication.py", line 348 in initialize
  File "UM/Application.py", line 288 in run
  File "UM/Qt/QtApplication.py", line 385 in run
  File "cura/CuraApplication.py", line 880 in run
  File "cura_app.py", line 243 in <module>

Extension modules: PyQt6.QtCore, PyQt6.QtNetwork, PyQt6.QtGui, PyQt6.QtWidgets, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg.lapack_lite, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, scipy._lib._ccallback_c, scipy.sparse._sparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._isolve._iterative, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg._cythonized_array_utils, scipy.linalg._flinalg, scipy.linalg._solve_toeplitz, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg.cython_lapack, scipy.linalg.cython_blas, scipy.linalg._matfuncs_expm, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.special._ellip_harm_2, scipy.spatial.transform._rotation, _cffi_backend, PyQt6.QtOpenGL, PyQt6.QtQml, PyQt6.QtQuick, pyArcus, PyQt6.QtSvg, pyclipper._pyclipper, pynest2d, scipy.optimize._minpack2, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize.__nnls, scipy.optimize._highs.cython.src._highs_wrapper, scipy.optimize._highs._highs_wrapper, scipy.optimize._highs.cython.src._highs_constants, scipy.optimize._highs._highs_constants, scipy.linalg._interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, shapely.lib, shapely._geos, shapely._geometry_helpers, scipy.ndimage._nd_image, _ni_label, scipy.ndimage._ni_label (total: 87)
[1]    73878 IOT instruction (core dumped)  QT_QPA_PLATFORM=xcb ./UltiMaker-Cura-5.8.1-linux-X64.AppImage --debug
`

**Comment by gigem** [comment]
This problem seams to be related to the libz3-4 package.  Cura works with version 4.8.12-3.1+b2 but not with version 4.8.12-3.1+b3 and later.  Because b3 is simply a rebuild with different tools and libraries, the real problem probably lies elsewhere but I've yet to find it.  My work around for now is to extract libz3.so.4 from the b2 package and put it in a custom LD_LIBRARY_PATH when running cura.

**Comment by wklaebe** [comment]
Unpacking the AppImage, dropping the libz3.so.4 from libz3-4_4.8.12-3.1_amd64.deb into there, and starting Cura with ./AppRun works too.

**Comment by rephlex** [comment]
> Where is the bug though? In libz3-4 or somewhere else?

Nah, it's just Cura being incompatible with a newer libz3-4 version, probably because of some function call change.

**Comment by julienrobin28** [comment]
Hello there,

Just as a feedback, I can confirm those points (on Debian 13/Trixie up to date as 2024-12-06)

 - When using `amdgpu` driver, be it on Radeon 760M IGP or R9 280X GPU, `UltiMaker-Cura-5.9.0-linux-X64.AppImage` give the described error about "Failed to finding matching FBConfig for QSurfaceFormat"

 - When disabling the 760M IGP and forcing the NVIDIA GPU, using proprietary NVIDIA driver, no more error: it works.

 - Back on amdgpu, having downloaded the previous libz3-4 version `libz3-4_4.8.12-3.1_amd64.deb` package for Debian 12/Bookworm (and having it extracted manually into `/some/folder/old-libz3-4/`) Cura works fine when launched with `LD_LIBRARY_PATH=/some/folder/old-libz3-4/ ./UltiMaker-Cura-5.9.0-linux-X64.AppImage`

Congrats to @gigem for having found that!

**Comment by vostok01** [comment]
I have no libz3 library in my gentoo distro at all, but this error is present. libz3 is missing both in my system and in UltiMaker-Cura-5.9.0-linux-X64.AppImage. Moreover, libz3 has never been installed at all.
Therefore, I don't really understand how Cura can depend on libz3.

**Comment by ls-2020** [comment]
> I have no libz3 library in my gentoo distro at all, but this error is present. libz3 is missing both in my system and in UltiMaker-[Cura-5](https://ultimaker.atlassian.net/browse/CURA-5).9.0-linux-X64.AppImage. Moreover, libz3 has never been installed at all. Therefore, I don't really understand how Cura can depend on libz3.

same error here and no libz3! my guess is that this bug is related to **amdgpu** and not to libz3

**Comment by rephlex** [comment]
> my guess is that this bug is related to **amdgpu** and not to libz3

I confirm this, as from multiple systems running the same package versions but various intel/nvidia/amd GPUs, only systems which use AMD GPUs are affected.

**Comment by vostok01** [comment]
> only systems which use AMD GPUs are affected.

I also have an AMD GPU.

**Comment by JeanLucCoulon** [comment]
I've a core i7-6700K with the integrated graphic attachment (Intel HD
Graphics 530/
And I've the problem.

Jean-Luc

**Comment by TheRealOne78** [comment]
Do you also have an AMD dedicated GPU, or just the integrated GPU?

> I've a core i7-6700K with the integrated graphic attachment (Intel HD
> Graphics 530/
> And I've the problem.

**Comment by JeanLucCoulon** [comment]
Nope

Le mer. 18 déc. 2024 à 18:33, TheRealOne78 ***@***.***> a
écrit :

**Comment by ls-2020** [comment]
@t-cc
please change the issue to:  **[5.x] AppImage fails to start on LINUX**

If **MESA** is compiled with Rusticl the AppImage fails to start, so many distributions are affected!

You can  downgrade (no nice solution) mesa to an older version and the AppImage will start again.

To fix this issue the code that initializes GLX should be updated to work with latest release of mesa.

**Comment by ask6483** [comment]
I can confirm the problem on Debian trixie (testing) with Intel HD Graphics 620.
I also tried the fix mentioned here with older version of libz3 from package `libz3-4_4.8.12-3.1_amd64.deb`, this doesn't work.

**Comment by ask6483** [comment]
I managed to build Cura from source and run it successfully. 

How to create appimage? Documentation is lacking support on this topic.

I have tried to use `crate_appimage.py` but I don't know what to pass via `dist_path` parameter.

**Comment by kostnik** [comment]
same here on gentoo, cura 5.9.0 (https://github.com/BlueManCZ/edgets/blob/master/media-gfx/cura/cura-5.9.0.ebuild):

```sh
$ cura
Gtk-Message: 19:12:30.444: Failed to load module "xapp-gtk3-module"
Gtk-Message: 19:12:30.444: Failed to load module "canberra-gtk-module"
Fontconfig warning: "/etc/fonts/conf.avail/05-reset-dirs-sample.conf", line 6: unknown element "reset-dirs"
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
Could not initialize GLX
Aborted
```

**Comment by theli-ua** [comment]
Just can't make it work here on gentoo on amd now at all. nor downgrading z3 nor mesa helped. Not sure how to workaround :(

**Comment by wjskrydl** [comment]
> Just can't make it work here on gentoo on amd now at all. nor downgrading z3 nor mesa helped. Not sure how to workaround :(

@theli-ua Here is the shell script I use to fix the AppImage on Gentoo... You will need [appimagetool](https://github.com/AppImage/appimagetool) to repackage it. Not sure if replacing libstdc++ is still required, but I had an issue with it in the past. I have been having issues with the Cura AppImage since early 5.x.x, replacing libwayland-client was the original fix. Hope this helps.
```bash
#!/bin/bash
if [[ $# -ne 1 ]]; then
        echo "usage: ./fix.sh [UltimakerAppImage]" 1>&2
        exit 1
fi

# Extract AppImage contents
./$1 --appimage-extract

# Copy libwayland-client.so from host
cp -H /usr/lib64/libwayland-client.so ./squashfs-root

# Grab GCC version from eselect
gcc_ver=$(eselect gcc list | cut -d '-' -f 5 | cut -d ' ' -f 1)
# Copy libstdc++.so.6 from host
cp -H /usr/lib/gcc/x86_64-pc-linux-gnu/$gcc_ver/libstdc++.so.6 ./squashfs-root/runtime/compat/

# Fix icon name issue
cat ./squashfs-root/com.ultimaker.cura.desktop | grep -v "Icon=cura-icon.png" > com.ultimaker.cura.desktop
echo "Icon=cura-icon" >> com.ultimaker.cura.desktop
mv com.ultimaker.cura.desktop squashfs-root/

# Repack the AppImage. AppImageTool required.
./appimagetool/appimagetool-x86_64.AppImage squashfs-root/
```

**Comment by fmgrotepass** [comment]
Crap! This problem returned for me. I'm running on Gentoo with an Intel onboard graphics card. Last time I solved it by downgrading  Mesa, but now things broke again. The image fix shared by @wjskrydl  did not work.

I did some tests and dug around. Found the following and shared it on the Gentoo forums:

https://forums.gentoo.org/viewtopic-p-8863541.html#8863541

Tomorrow I'll try some of the possible solutions

Updates shared on the Gentoo forum, but in short, downgrading Mesa at this point is not that simple. I tried some of the suggestions to export QT_XCB_GL_INTEGRATION=none but somehow logically this just nukes the GL abilities and produces a popup that Cura needs this and will not work without it. With the problem reported in the FBConfig part of qt_qlx I think that it is related to Mesa and the like. The test code shared on the Gentoo forum does run without problems. I'll look into building Mesa with extra debug and to add MESA_DEBUG to my env when running. Maybe things will be clearer then. My system runs with Mesa 25.0.5 and the latest Ubuntu (most common Linux choice) is on 24.3.4, so if it is caused by Mesa 25, then it will not manifest in 
most of the systems

**Comment by fmgrotepass** [comment]
Tested on all versions of Cura from 5.6.0 up to 5.10.1 and the problem persists. This problem was also reported on other forums for users using FreeCAD, but those AppImages launch according to expectation on my machine.

**Comment by fmgrotepass** [comment]
I solved the problem for my machine. I did change 2 things at the same time: 

The one that I think solved it was disabling the llvm use flag for mesa. I also enabled the z3 flag (as it was also mentoined here) for llvm. Maybe if I have time I'll see which one causes the bug, but for now it works and the long queue of print jobs that need printing can finally be done! Gentoo Linux users have a solution how to tackle this and if newer versions of Ubuntu or other Linux distro's have this issue, then at least there is some information for a solution.

**Comment by Liriel** [comment]
Can confirm: compiling mesa without `llvm` does the trick for me as well. Tested with `UltiMaker-Cura-5.9.0-linux-X64.AppImage` on Gentoo with `mesa-25.0.7`and `llvm-19.1.7`.

**Comment by m-sundman** [comment]
This works for me:
`LD_LIBRARY_PATH=/folder/containing/libz3.so.4/file ./UltiMaker-Cura-5.10.1-linux-X64.AppImage`

**Comment by mlepage** [comment]
Works for me too. To summarize (for those who might have trouble implementing the workaround):

- Download bookworm deb from one of the sites listed at https://packages.debian.org/bookworm/amd64/libz3-4/download
- Verify the contents with a command like: `dpkg --contents libz3-4_4.8.12-3.1_amd64.deb`
- Extract the deb with a command like: `dpkg-deb -x libz3-4_4.8.12-3.1_amd64.deb libz3-4_4.8.12-3.1_amd64`
- Figure out the path to where libz3.so.4 has been extracted, for the next step
- Run the AppImage with a command like `LD_LIBRARY_PATH=~/Downloads/libz3-4_4.8.12-3.1_amd64/usr/lib/x86_64-linux-gnu ./UltiMaker-Cura-5.10.2-linux-X64.AppImage` (where LD_LIBRARY_PATH contains libz3.so.4)

_This is only a workaround until the AppImage is fixed, of course._

**Comment by kylegordon** [comment]
I just want to chime in and say to other casual users such as myself that the LD_LIBRARY_PATH parameter needs to be the path to the directory in which libz3.so.4 exists, not the file itself.

in my case, LD_LIBRARY_PATH=~/Downloads/libz/usr/lib/x86_64-linux-gnu/

**Comment by cdp1337** [comment]
+1 to this issue too; just upgraded to Trixie and went to print something I was working on, only to realize the AppImage didn't load with that GLX error.

Didn't try the LD_LIBRARY_PATH work-around, but `sudo apt install cura` installed v5.0.0 from the repos, which seems to work well enough, (providing you don't need the latest features).

**Comment by JeanLucCoulon** [comment]
>> just upgraded to Trixie

I've trixie too (sid in fact).
I've got libz3.so.4 from libz3-4_4.8.12-3.1_amd64.deb and put it in
/usr/local/lib
set the path accordingly.
no problems so far.

Jean-Luc

Le dim. 17 août 2025 à 10:04, Charlie Powell ***@***.***> a
écrit :

**Comment by theli-ua** [comment]
Still can't make this to work on Gentoo, used to work if I recompile mesa without llvm, now it won't even let me do it as it'll  disable radeonsi driver

**Comment by theli-ua** [comment]
This looks to be simply because ultimaker uses very old ubuntu runner to build the appimage and as distros update the more and more stop working with these appimages (Are there even any more working left??). Either way, I've bumped runner in github workflow to ubuntu24.04, and an appimage that it built works fine for me 
https://github.com/theli-ua/Cura/releases/tag/ubuntu24
https://github.com/theli-ua/Cura/actions/runs/17533624102/job/49793805491

**Comment by davetheghost** [comment]
> This looks to be simply because ultimaker uses very old ubuntu runner to build the appimage and as distros update the more and more stop working with these appimages (Are there even any more working left??). Either way, I've bumped runner in github workflow to ubuntu24.04, and an appimage that it built works fine for me https://github.com/theli-ua/Cura/releases/tag/ubuntu24 https://github.com/theli-ua/Cura/actions/runs/17533624102/job/49793805491

gracias, amigo

was giving me crap too, AMD GPU related it appears

I mean, i do understand using old software for compatibility reasons... but when the application begins falling apart... wake up!

**Comment by jimy-byerley** [comment]
same problem here on debian 13, @theli-ua 's built did the job

**Comment by martini1992** [comment]
Also having this issue after upgrading to LMDE 7 which is based on Debian 13, Intel iGPU.

UM/Settings/SettingFunction.py:244: DeprecationWarning: ast.Str is deprecated and will be removed in Python 3.14; use ast.Constant instead
  def visit_Str(self, node: ast.Str) -> None:
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
qt.glx: qglx_findConfig: Failed to finding matching FBConfig for QSurfaceFormat(version 4.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize -1, redBufferSize 1, greenBufferSize 1, blueBufferSize 1, alphaBufferSize -1, stencilBufferSize -1, samples -1, swapBehavior QSurfaceFormat::SingleBuffer, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::CoreProfile)
Could not initialize GLX
Aborted (core dumped)

**Comment by alberic89** [comment]
Same problem, I made a fixed AppImage using the following steps:

* Get the libz3 shared lib using this method:
>     * Download bookworm deb from one of the sites listed at https://packages.debian.org/bookworm/amd64/libz3-4/download
> 
>     * Verify the contents with a command like: `dpkg --contents libz3-4_4.8.12-3.1_amd64.deb`
> 
>     * Extract the deb with a command like: `dpkg-deb -x libz3-4_4.8.12-3.1_amd64.deb libz3-4_4.8.12-3.1_amd64`
> 
>     * Figure out the path to where libz3.so.4 has been extracted, for the next step

* Get the Cura Appimage and extract it by running:
```bash
./UltiMaker-Cura-5.11.0-linux-X64.AppImage --appimage-extract
```
All the content is extracted to a directory named `squashfs-root`
* Copy the `libz3.so.4` file in this directory
* Edit the `squashfs-root/com.ultimaker.cura.desktop` file to remove the `.png` of the icon name. Otherwise, the re-build will fail.
* Download the appimage-builder executable at https://github.com/AppImageCrafters/appimage-builder/releases/latest
* Run the following command:
```bash
ARCH=x86_64 ./appimagetool-x86_64.AppImage squashfs-root
```
This is all assuming you are using a x86_64 architecture.
The result (available for 45 days): https://drop.infini.fr/r/fZbo6m5utj#T8n2wtcFKjj7z3OD5QM0SkBTh4e0Rm8txDuOY4FggFc=

**Comment by jeremywang19740412** [comment]
> This problem seams to be related to the libz3-4 package. Cura works with version 4.8.12-3.1+b2 but not with version 4.8.12-3.1+b3 and later. Because b3 is simply a rebuild with different tools and libraries, the real problem probably lies elsewhere but I've yet to find it.

Man, you are amazing, how did you find out, me can not figure this out myself until I see your solution (me even looked at the coredump file but no clue), thank you really~

**Comment by gigem** [comment]
On Sun, Nov 09, 2025 at 11:46:40PM -0800, Jeremy Wang wrote:
> jeremywang19740412 left a comment (Ultimaker/Cura#19837)
> 
> > This problem seams to be related to the libz3-4 package. Cura works with version 4.8.12-3.1+b2 but not with version 4.8.12-3.1+b3 and later. Because b3 is simply a rebuild with different tools and libraries, the real problem probably lies elsewhere but I've yet to find it.
> 
> Man, you are amazing, how did you find out, me can not figure this out myself until I see your solution (me even looked at the coredump file but no clue), thank you really~

It's been so long that I don't remember the specifics.  As best as I
can recall, older versions of Cura suddenly failed too so the cause
was likely something that changed on my system and dependent libraries
were the logical things to try first.

David
-- 
David Engel
***@***.***

**Comment by HakanKoseoglu** [comment]
< tap tap> Is this project on? How on earth no one is picking this up and fixing it?

**Comment by davetheghost** [comment]
> < tap tap> Is this project on? How on earth no one is picking this up and fixing it?

there's over a thousand issues-- also, prob for compatibility reasons

**Comment by HakanKoseoglu** [comment]
> > < tap tap> Is this project on? How on earth no one is picking this up and fixing it?
> 
> there's over a thousand issues-- also, prob for compatibility reasons

Yeah, this definitely fixes the compatibility problems by rendering it completely useless for thousands of people.

**Comment by rephlex** [comment]
> > < tap tap> Is this project on? How on earth no one is picking this up and fixing it?
> 

It's probably one of
- WoRkS oN ArChLiNuX, wOnTfIx!!11
- nObOdY iS uSiNg DeBiAn AnYwAyS
- nObOdY iS uSiNg LuNiX AnYwAyS

**Comment by rephlex** [comment]
same as #18891 #19837 #19913 #20353 #20706 #20806 #21089 #20516

**Comment by davetheghost** [comment]
https://stuff.daves.repair/UltiMaker-Cura-5.11.0-linux-X64.AppImage

This should work for Trixie

All I did was replace libz3 and repack.

**Comment by ahmouse15** [comment]
> https://stuff.daves.repair/UltiMaker-Cura-5.11.0-linux-X64.AppImage
> 
> This should work for Trixie
> 
> All I did was replace libz3 and repack.

Thanks for sharing. Interestingly enough, I still get the same error even with your build, on Gentoo Linux.

**Comment by shivajiva101** [comment]
repacking with libz3.so.4 still works with Debian 13

**Comment by csirmaz** [comment]
Here are detailed instructions that worked for me to repack the AppImage with libz3.so.4 on Debian 13: https://www.epcsirmaz.co.uk/tech_blog/2026-03-appimage-recompile-cura.html

**Comment by Rpnpif** [comment]
Thanks to @csirmaz. This workaround works fine.

But we expect that this issue will be really fixed for Debian Trixie.
